src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div
+    v-if="!isBadge"
+    class="k-copy"
+    :class="truncate ? null : 'copy-truncated'"
+  >
+    <KTooltip
+      v-if="format !== 'hidden'"
+      :class="truncate ? null : 'copy-content'"
+      data-testid="copy-content"
+      :label="contentTooltip"
+      placement="bottomStart"
+      position-fixed
+    >
+      <div
+        :class="[
+          'content-container',
+          truncate ? 'truncated-content' : 'non-truncated-content',
+          monospace ? 'monospace' : null
+        ]"
+        :style="widthStyle"
+      >
+        {{ contentFormat }}
+      </div>
+    </KTooltip>
+
+    <KTooltip
+      class="content-icon-wrapper"
+      :label="`${props.badgeText}` ? iconTitle : tooltipText"
+      max-width="500px"
+      placement="bottomStart"
+      position-fixed
+    >
+      <KClipboardProvider v-slot="{ copyToClipboard }">
+        <span
+          data-testid="copy-to-clipboard"
+          role="button"
+          tabindex="0"
+          @click.stop="copyIdToClipboard(copyToClipboard)"
+        >
+          <KIcon
+            class="content-icon"
+            :hide-title="!!copyTooltip || undefined"
+            icon="copy"
+            size="16"
+            :title="iconTitle"
+          />
+        </span>
+      </KClipboardProvider>
+    </KTooltip>
+  </div>
+
+  <div
+    v-else
+    class="k-copy-badge"
+  >
+    <span
+      v-if="isBadge"
+      class="copy-badge-text"
+    >
+      {{ badgeText }}
+    </span>
+    <KBadge
+      v-if="isBadge"
+      appearance="custom"
+      background-color="#f9fafb"
+      border-color="#d7d8fe"
+      class="id-badge"
+      color="#473cfb"
+      is-bordered
+      shape="rectangular"
+    >
+      <KCopy
+        :badge-text="badgeText"
+        class="id-badge-content"
+        :content="content"
+        :content-tooltip="contentTooltip"
+        :copy-tooltip="copyTooltip"
+        :monospace="monospace"
+        :success-tooltip="successTooltip"
+        :truncate="truncate"
+        :truncation-limit="truncationLimit"
+      />
+    </KBadge>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+import { computed, ref } from 'vue'
+import KBadge from '@/components/KBadge/KBadge.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KClipboardProvider from '@/components/KClipboardProvider'
+import KTooltip from '@/components/KTooltip/KTooltip.vue'
+
+const props = defineProps({
+  // text displayed before the copyable content when `isBadge` is true
+  badgeText: {
```

Comment:
question: should this be "label" or something instead? Maybe I'm just confused.

Also, why does the component need to provide the label? Couldn't the host app just choose to display text before the component?

Reply:
I'm good with not calling it `label` then; however, I'm still not sure I see the need for this within the component.

Reply:
Ah missed your point there but now I agree - seems redundant since host app can take care of that. Ultimately if we want to make sure styling is always the same in text used in front of the badge variant - I would suggest making this a default slot and adding styling on a wrapping container.

Reply:
Going to add a `slot` with name `badgeText` to fulfil this requirement: `copyTooltip - tooltip text displayed on hover over copy button (default: if `badgeText` provided “Copy <badge-text>” else “Copy”)`

Reply:
@kaiarrowood I am not sure how to display `Copy <badge-text>` if there is no prop for badgeText. Can you clarify if we need that prop or not ?

Reply:
This absolutely needs to be a prop and not a slot. We use the `badge-text` prop for both the label and the default copy tooltip text. I don't care what we call the prop but but there is no point adding a slot when in most use cases we intend to use this text in the copy tooltip also.

Reply:
Maybe I'm confused, but are we talking about the text that comes before the copyable content?

```md
My label: 123456789 []
```

In this example ^ "My label" is the text I'm referring to. Do we really need a prop/slot for this? I'm not sure I understand why the host app can't just put text before the element if desired? Sorry in advance if I'm missing something

Reply:
The tooltip text on hover needs to display that text 'Copy My Label' instead of 'Copy'; if there is no prop how can that be extracted ?

Reply:
Correct, we have logic to scrape the `:` off whatever is passed for `badge-text` (the label) and also use it in the tooltip displayed on hover of the copy button `Copy <badge-text>`